### PR TITLE
docs: Add initial version of ecosystem page in user guide

### PR DESCRIPTION
**Pull request by r-brink**
Related to this issue: https://github.com/pola-rs/polars/issues/13778

### CODE REVIEW

File: docs/user-guide/ecosystem.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Polars ecosystem
+
+## Introduction
+
+On this page you can find a non-exhaustive list of libraries and tools that support Polars. As the data ecosystem is evolving fast, more libraries will likely support Polars in the future. One of the main drivers is that Polars makes use of `Apache Arrow` in it's backend.
+
+#### [Apache Arrow](https://arrow.apache.org/)
+
+Apache Arrow enables zero-copy reads of data, meaning that data can be directly accessed in its in-memory format without the need for copying or serialisation. This greatly enhances performance, especially when sharing data between different processes or systems. Using Apache Arrow, Polars is compatible with a wide range of libraries that also make use of Apache Arrow, like Pandas and Pyspark.
```
Comment (ritchie46) [inline docs/user-guide/ecosystem.md:9]: This is only possible within the same process without copying. It is true that there is no serializatin if two differen processes share arrow memory. 

> This greatly enhances performance

I would mostly say that it enhances performance when integrating with different tools.

File: docs/user-guide/ecosystem.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+hvPlot is available as the default plotting backend for Polars making it simple to create interactive and static visualisations. You can use hvPlot by using the feature flag `plot` during installing.
+
+```python
+pip install 'polars[plot]'
+```
+
+#### [Matplotlib](https://matplotlib.org/)
+
+Matplotlib is a comprehensive library for creating static, animated, and interactive visualizations in Python. Matplotlib makes easy things easy and hard things possible.
```
Comment (ritchie46) [inline docs/user-guide/ecosystem.md:23]: How does polars interact with matplotlib?

Comment (r-brink) [reply docs/user-guide/ecosystem.md:23]: There is a pending PR for a page about data visualisations from @braaannigan . My idea is to link to that page or show the plots on this page. I have opened an issue to open the discussion. That is also why this PR is in draft

File: docs/user-guide/ecosystem.md (hunk 1 above)
Comment (r-brink) [reply docs/user-guide/ecosystem.md:9]: Thanks, I'll update in the next iteration

File: docs/user-guide/ecosystem.md (hunk 2 above)
Comment (braaannigan) [reply docs/user-guide/ecosystem.md:23]: @r-brink I'll try to get my PR fixed in the next couple of days

Comment (r-brink) [reply docs/user-guide/ecosystem.md:23]: @braaannigan that would be great. I will update the PR with links to the section on your visualisation page!

Comment (braaannigan) [reply docs/user-guide/ecosystem.md:23]: @r-brink Hopefully can get this merged soon #13052

**Pull request closed without merging**